lib/minitest/silence_plugin.rb

```diff
@@ -37,6 +37,7 @@ def run_one_method(klass, method_name)
         end
 
         result.output = output_thread.value
+        output_reader.close
```

Comment:
Shouldn't this be in the ensure block after `output_writer.close` so that it is closed even if `super` raises?

Reply:
I assumed that there would be some sort of thread access or read-after-close issue, since `output_reader.read` is called within a separate thread, and `.value` is called on that thread after the ensure block.

I'm not totally familiar with IO and threading in Ruby, but I ran a quick test. Having the output_reader.close call in the ensure block seems to result in an error as I expected:
```ruby
...
ensure
          $stdout.reopen(@original_stdout)
          $stderr.reopen(@original_stderr)
          $stdin.reopen(@original_stdin)
          output_writer.close
          output_reader.close
        end

        result.output = output_thread.value
end
```

```ruby
Traceback (most recent call last):
	1: from /Users/adamshepley/.gem/ruby/2.7.2/gems/minitest-silence-0.2.1/lib/minitest/silence_plugin.rb:24:in `block in run_one_method'
/Users/adamshepley/.gem/ruby/2.7.2/gems/minitest-silence-0.2.1/lib/minitest/silence_plugin.rb:24:in `read': stream closed in another thread (IOError)
```

Willem mentioned a different approach in slack:
> what I always wanted to do is find a way to re-use the same pipe over and over again for every test, but I haven’t gotten around to do that.

But I am not familiar enough with Ruby IO to quickly make that change in a correct manner, and deferred to a simpler change that at least improved the situation even if it wasn't a full fix.

Reply:
For the purposes of safety, an alternative option may be an ensure block for the entire method:

```ruby
result = begin
    $stdout.reopen(output_writer)
    $stderr.reopen(output_writer)
    $stdin.reopen(File::NULL)

    super
  ensure
    $stdout.reopen(@original_stdout)
    $stderr.reopen(@original_stderr)
    $stdin.reopen(@original_stdin)
    output_writer.close
  end

  result.output = output_thread.value
  result
ensure
  output_reader.close
end
```
It's not as neat as keeping it in the same block, but looks like it works fine?

Reply:
Oh right, we do need to do it after `output_thread.value`.  I wasn't familiar with Thread#value, so didn't realize that was where Thread#join was effectively being called.  :+1: to closing in the method's ensure block